metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/common/shell/StellarShell.java

```diff
@@ -308,17 +315,68 @@ private void handleMagic( String rawExpression) {
               .collect(Collectors.joining(", "));
       writeLine(functions);
 
-    } else if(MAGIC_VARS.equals(command)) {
+    } else if (MAGIC_VARS.equals(command)) {
 
-      // list all variables
+      // '%vars' -> list all variables in scope
       executor.getVariables()
-              .forEach((k,v) -> writeLine(String.format("%s = %s", k, v)));
+              .forEach((k, v) -> writeLine(String.format("%s = %s", k, v)));
+
+    } else if (MAGIC_GLOBALS.equals(command)) {
+
+      // '%globals' -> list all globals in scope
+      Map<String, Object> globals = Collections.emptyMap();
```

Comment:
I don't understand why we check for capabilities here, but not when setting.  It looks like I can define as much as I want, but only see if the capability is there?  Why do we only guard here?

Reply:
Why wouldn't we getOrCreate here as well?

Reply:
It is just not strictly needed since if I start out with an empty map and there is no global config in Zk, then we are just left with the empty map, so we're good.  So functionally getOrCreate is not needed here.

That being said, I can see how it would bring a nice sense of symmetry to just use getOrCreate in this case also.  Let me try that out real quick.

Reply:
Yep, it is prettier this way.  Thanks  @ottobackwards

Reply:
When would you not get an empty map?  It looks like if you are not using ZK you always get an empty map.  I must be missing something

Reply:
Never mind, calling getOrCreate will put the capability in. I see. Sorry

Reply:
Yep, exactly.  No worries.  I'm glad you're looking at it closely.

Reply:
Nick. If I understand correctly, if there is no ZK, then this turns on the GLOBAL_CONFIG flag, and sets whatever properties.

Is there any risk that other functions that check for GLOBAL_CONFIG can see the flag, but assuming the full config is present, when in this case only what the user has set is present?

Reply:
I do not see a problem here.  Functions need whatever GLOBAL_CONFIG settings they need.  This PR does not make it harder or less likely for functions to get the values they need.  This PR makes it easier for a user to provide the settings, or alter the settings that a function needs.  IMHO